EvolutionModel/Foodbag.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Evolution.Model {
+    public class Foodbag {
+        private int foodamount;
+
+        public Foodbag(){
+            foodamount = 0;
+        }
+
+        public int Count => foodamount;
+
+        public void AddFood(int erbij){
```

Comment:
Tijd voor kommaneuken. Kan erbij 0 of negatief zijn? Is dat wenselijk? Naast `int` bestaat er ook `uint` waar de `u` staat voor `unsigned`.

Reply:
als ik van erbij een uint maak begint hij daarna te zeuren dat foodamount+=erbij niet kan want 'cannot implicitly convert type 'uint' to 'int'. An explicit conversion exists (are you missing a cast?) '

Reply:
fixed it! moest natuurlijk van foodamount zelf ook een uint maken

Reply:
Hmm, dat komt omdat uint.MaxValue > int.MaxValue. M.a.w., er is geen garantie dat een uint in een int past. Twee opties:

- Foodamount ook uint maken
- Erbij een int houden en zelf checken of erbij niet negatief is.

Het verschil hiertussen zou uit kunnen maken als het redelijkerwijs mogelijk was dat erbij te groot zou worden voor int, maar wel in uint zou passen. Dat is echter zo veel groter dan wat we voor deze toepassing zouden verwachten dat het in praktijk niet zo veel verschil maakt.


-----

Het verschil tussen int en uint is dat uint geen bit (0/1) hoeft te gebruiken om aan te geven of het getal wel of niet negatief is. Dit betekent dat er een extra bit beschikbaar is voor het opslaan van het getal zelf, en uint dus tot ~4.2 10^9 kan hebben ipv ~2.1 10^9.